sourcegraph/cody
# plg: display errors when autocomplete rate limits trigger

Closes #2134 

This bug has 2 parts to it:
1. Using the rate limited demo accounts, by design the headers `retry-after` return back the same time as when the request was made. This causes the error messages to constantly be displayed in the quick pick view
2. Also using the rate limited demo account, if you hard code the retry time to a later time (like 1 day after), the error message does not display anymore after being dismissed once. 


This PR adds a check to see whether or not the errors have been added in the `statusBar` and to display the rate limited error if no other `RateLimitError` is present. This also makes sure the error is only displayed once, instead of spamming the user if autocomplete keeps getting triggered.

**Before (bug 1):**

https://github.com/sourcegraph/cody/assets/51868853/de5a963a-a5be-4677-a75c-4f154960a516

**Before (bug2, after hardcoding a later `retry-after` date):**

https://github.com/sourcegraph/cody/assets/51868853/ffcd7748-0827-4fca-b71f-e141112ead97


After: 

https://github.com/sourcegraph/cody/assets/51868853/48078429-59dc-4ad8-a744-efb1e665a689



## Test plan
- login to `https://sourcegraph.com` using the free demo account ([thread](https://sourcegraph.slack.com/archives/C066F85TE80/p1701712911448229?thread_ts=1700694252.150349&cid=C066F85TE80))
- go to any file, trigger autocompletes multiple times and you will see only 1 error message displayed
- click on the error, dismiss it, and trigger another autocomplete. you should see another error message displayed
